lib/modules/settings/screens/current_location_picker.dart

```diff
@@ -70,41 +94,36 @@ class _CurrentLocationPickerState extends State<CurrentLocationPicker> {
         userLocation, userSettingsState.calculationMethod));
   }
 
-  Widget _getUserLocationString(UserLocation? location) {
-    if (location != null &&
-        location.cityAddress != null &&
-        location.country != null) {
-      return Text(
-        "${location.cityAddress}, ${location.country}",
-        style: Theme.of(context).textTheme.headline6,
-      );
-    } else {
-      return Container();
-    }
-  }
-
-  _updateUserLocation() async {
+  _getCoordinatesFromAddress(BuildContext context) async {
     if ((cityValue.isNotEmpty || stateValue.isNotEmpty) &&
         countryValue.isNotEmpty) {
-      Future<List<Location>> locations = locationFromAddress(
-          cityValue + ', ' + stateValue + ', ' + countryValue);
-      Location location = await locations.then((value) => value[0]);
-      context.read<UserSettingsBloc>().add(
-            UserLocationEvent(
-              UserLocation(
-                city: cityValue.isNotEmpty ? cityValue : null,
-                state: stateValue.isNotEmpty ? stateValue : null,
-                country: countryValue,
-                location: Location(
-                    latitude: location.latitude,
-                    longitude: location.longitude,
-                    timestamp: location.timestamp),
-              ),
-            ),
-          );
+      var address = "$cityValue, $stateValue, $countryValue";
+      try {
+        Location location = await _geoCodingAddress(address);
+        _updateUserLocation(context, location.latitude, location.longitude);
+        return;
+      } catch (e) {
+        debugPrint(e.toString());
+      }
+
+      var response = await getCoordinatesFromAddress(address);
+      var location = json.decode(response.body);
+      _updateUserLocation(context, location['latitude'] as double,
+          location['longitude'] as double);
```

Comment:
In here, we take the values of "country", "state",.... Then get the lat-lon, then use those to retrieve the "country", "state", values from Placemark API (in _updateUserLocation).
I think this unnecessary.

Reply:
To make use of [placemarkFromCoordinates](https://github.com/NedaaApp/nedaa/pull/15/files/bf9fab9f6a631e6b2de5994be08de34f4bc9258b#diff-8cf13d88a7f89dd55e33e3d3512e50b7092aaac40cf9a0f9050986aac9ecdf17) to translate the country, state and city based on the current locale